include/graphqlservice/GraphQLService.h

```diff
@@ -686,62 +675,44 @@ struct ModifiedResult
 				{
 					for (auto& entry : wrappedResult)
 					{
-						children.push(ModifiedResult::convert<Other...>(std::move(entry),
+						children.push_back(ModifiedResult::convert<Other...>(std::move(entry),
 							ResolverParams(wrappedParams)));
 						++std::get<size_t>(wrappedParams.errorPath.back());
 					}
 				}
 
-				response::Value data(response::Type::List);
-				response::Value errors(response::Type::List);
+				ResolverResult document { response::Value { response::Type::List } };
 
 				wrappedParams.errorPath.back() = size_t { 0 };
 
-				while (!children.empty())
+				for (auto& child : children)
 				{
 					try
 					{
-						auto value = children.front().get();
-						auto members = value.release<response::MapType>();
+						auto value = child.get();
 
-						for (auto& entry : members)
-						{
-							if (entry.second.type() == response::Type::List
-								&& entry.first == strErrors)
-							{
-								auto errorEntries = entry.second.release<response::ListType>();
+						document.data.emplace_back(std::move(value.data));
 
-								for (auto& errorEntry : errorEntries)
-								{
-									errors.emplace_back(std::move(errorEntry));
-								}
-							}
-							else if (entry.first == strData)
+						if (!value.errors.empty())
+						{
+							document.errors.reserve(document.errors.size() + value.errors.size());
+							for (auto& error : value.errors)
 							{
-								data.emplace_back(std::move(entry.second));
+								document.errors.push_back(std::move(error));
```

Comment:
since you're changing the types, worth looking into making `errors` a list, so you can `splice()`, which is simpler (1 line) and cheaper (although errors are not the hot path).

Reply:
For large enough nodes, a `std::list` may be worth it.  But otherwise you do pay a premium in memory usage per node to store the extra next/prev link pointers. It's also bad from a data locality/cache perspective since every node is allocated separately, so I generally avoid using them. Some more background here: https://www.stroustrup.com/bs_faq.html#list.

Reply:
yeah, I get the difference between list/vector and even variations such as inlist (inline lists, where each element is a linked list node per se).

Really, the wasted amount or locality does not matter for handling errors, even more compared to the current situation where we're creating a new array and moving everything -- or before, where those were actually `response::Value` maps, with the map + array, strings getting copied. We already improved a lot, the move to list/splice would basically save a minimum while making the code simper (no more reserve/loop/...).

Reply:
That's true, propagating errors happens a lot more than the final packaging into a `response::Value`, and `std::list::splice` can do that in constant time.